lib/screens/profile_screens/profile/components/switch_account_bottom_sheet/interactor/usecases/import_accounts_use_case.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:async/async.dart';
+import 'package:seeds/datasource/local/settings_storage.dart';
+import 'package:seeds/datasource/remote/api/key_accounts_repository.dart';
+import 'package:seeds/datasource/remote/api/profile_repository.dart';
+
+class ImportAccountsUseCase {
+  final KeyAccountsRepository _keyAccountsRepository = KeyAccountsRepository();
+  final ProfileRepository _profileRepository = ProfileRepository();
+
+  Future<List<Result>> run(String publicKey) async {
+    final accountsResponse = await _keyAccountsRepository.getKeyAccountsMongo(publicKey);
+    if (accountsResponse.isError) {
+      final List<Result> items = [accountsResponse];
+      return items;
+    } else {
+      final List<String> currentKeyAccounts = accountsResponse.asValue!.value;
+      final List<String> savedAccounts = settingsStorage.accountsList;
+      // Remove duplicated accounts
+      savedAccounts.removeWhere((i) => currentKeyAccounts.contains(i));
```

Comment:
I think we should keep saved as is and add current (not remove current from saved, which changes the list order)

Anyway maybe I am wrong, but I think we should aim to preserve list order

Reply:
This is a good one, I thought that maybe a key could create a new account is not in local so that the local one is always updated.

> but I think we should aim to preserve list order
Well in this case I always prepend the current key accounts first.

Reply:
the order in which keys are in doesn't matter, that's not shown to users

but the order in which the accounts are in is shown to user, best preserve order